collector/lib/Logging.cpp

```diff
@@ -102,7 +102,20 @@ bool ParseLogLevelName(std::string name, LogLevel* level) {
 
 void InspectorLogCallback(std::string&& msg, sinsp_logger::severity severity) {
   auto collector_severity = (LogLevel)severity;
-  collector::logging::LogMessage(__FILE__, __LINE__, false, collector_severity) << msg;
+
+  if (msg.rfind("libbpf:", 0) == 0 && collector_severity == LogLevel::DEBUG) {
+    // downgrade libbpf debug logs to TRACE to avoid thousands of lines
+    // of verbose relocation logging
+    collector_severity = LogLevel::TRACE;
+  }
+
+  // remove any newlines to avoid additional empty log lines
+  // because our logging already appends a newline
+  msg.erase(std::remove(msg.begin(), msg.end(), '\n'), msg.cend());
+
+  if (collector::logging::CheckLogLevel(collector_severity)) {
```

Comment:
We might want to move this check to the top of the function, otherwise we'll be doing an `rfind` and `erase` on messages that might not be logged at all.